app/modules/updater.py

```diff
@@ -202,6 +202,54 @@ def updateBoot(self):
             log.debug("Copied " + src + " to " + dst)
         return True
 
+    def updateSupervisorConf(self):
+        """Updating the supervisor.conf on the target rootfs, so after reboot it
+        will have the correct values. The relevant variables are pulled in from
+        the configuration.
+        """
+        log.info("Started to upgrade supervisor conf...")
+        supervisor_image = getConfigurationItem(self.conf, 'Supervisor', 'supervisor_image')
+        supervisor_tag = getConfigurationItem(self.conf, 'Supervisor', 'supervisor_tag')
+        if (not supervisor_image) or (not supervisor_tag):
+            log.debug('No supervisor conf is performed as no supervisor info is passed.')
+        else:
+            supervisorConf = os.path.join(self.tempRootMountpoint, 'etc/supervisor.conf')
+            tmpSupervisorConf = "/tmp/supervisor.conf"
+            try:
+                added_image, added_tag = False, False
+                with open(tmpSupervisorConf, "w") as tmpfile:
+                    with open(supervisorConf, "r") as supervisorconffile:
+                        for line in supervisorconffile:
+                            # Find and replace supervisor config values
+                            if re.match("^SUPERVISOR_IMAGE=.*$", line):
+                                log.debug("Replacing: SUPERVISOR_IMAGE=" + supervisor_image)
+                                print("SUPERVISOR_IMAGE={}".format(supervisor_image), file=tmpfile)
+                                added_image = True
+                            elif re.match("^SUPERVISOR_TAG=.*$", line):
+                                log.debug("Replacing: SUPERVISOR_TAG=" + supervisor_tag)
+                                print("SUPERVISOR_TAG={}".format(supervisor_tag), file=tmpfile)
+                                added_tag = True
```

Comment:
@willnewton I see, you mean not "replace or add", just "add" and then the tags will be always appended to the end of the supervisor.conf, right?